router/pkg/cors/config.go

```diff
@@ -17,7 +17,8 @@ type cors struct {
 }
 
 var (
-	DefaultSchemas = []string{
+	maxRecursionDepth = 10 // Safeguard against deep recursion
```

Comment:
I think 10 probably makes much more sense!

Reply:
If necessary, we can make it configurable, but there doesn't seem to me to be a need for that at this point

Reply:
agreed